classifier_and_summarizer/summarizer/text_media_matcher.py

```diff
@@ -0,0 +1,173 @@
+''' Text Media Matcher
+This module contains utilities for matching
+text and media .It uses the gale shapley algorithm
+for finding the stable matching - and it is optimal
+for the preference of the media.
+This script contains the following classes:
+    *ElementWithIndex : represent the text along with index
+    *TextMediaMatcher : implements the main algorithm for matching
+                        returns the stable matching
+'''
+from scipy.spatial.distance import cosine as cosine_distance
+from sentence_transformers import SentenceTransformer
+
+
+class ElementWithIndex:
+    ''' Pairs the element(summary sentence/image description)
+    with the index of its occurence in the webpage
+    '''
+
+    def __init__(self, element_index, element_text):
+        self.element_index = element_index
+        self.element_text = element_text
+
+
+class TextMediaMatcher:
+    ''' Class containing utilties to find the Text-Media Matching
+    Finds the stable matching by applying the Gale Shalpley algorithm
+    It accepts a list of objects of type ElementWithIndex
```

Comment:
Eventually we would want to get this from Contents right? The conversion from Contents to using this - do you want to do it here or in some other class which will eventually call TextMediaMatcher (both approaches are fine I guess)

Reply:
I would prefer to have some other class call text media matcher - that class would contain the pre processing step and would also call the text media matcher. Hope this is alright!